scheduler.py

```diff
@@ -222,6 +251,81 @@ def get_worker_round_robin_mode(self, gpuEnabledTask):
         with self.workerMapLock:
             return self.workers[pickedWorkerId]
 
+    def get_worker_power_of_two(self, gpuEnabledTask):
+        if gpuEnabledTask is True:
+            wrappedWorkerInfo =  self.get_worker_info_power_of_two(
+                self.WorkersPowerOf2WithGPU,
+                self.WorkersPowerOf2WithGPULock,
+                self.WorkersListPowersOf2WithGPU)
+            
+            if wrappedWorkerInfo is None:
+                logging.info(f"No worker found.")
+                raise NoWorkerAvailableError(self.schedulerMode)
+            
+            logging.info(f"Received worker {wrappedWorkerInfo} for GPU Enabled Task.")
+            return wrappedWorkerInfo
+        else:
+            if len(self.WorkersListPowersOf2WithoutGPU) == 0:
+                # If no workers without gpu - try getting from worker with gpu
+                wrappedWorkerInfo =  self.get_worker_info_power_of_two(
+                    self.WorkersPowerOf2WithGPU,
+                    self.WorkersPowerOf2WithGPULock,
+                    self.WorkersListPowersOf2WithGPU)
+
+                if wrappedWorkerInfo is None:
+                    logging.info(f"No worker found.")
+                    raise NoWorkerAvailableError(self.schedulerMode)
+                
+                logging.info(f"No non GPU workers found. Got one GPU worker {wrappedWorkerInfo} for non GPU Enabled Task.")
+                return wrappedWorkerInfo
```

Comment:
Suggestion: Can call `get_worker_power_of_two(gpuEnableTask = true)` here instead of duplicate logic.

Reply:
Discussed offline, this is lower priority. Will update in follow up PR later.